app/kudos/templates/share.html

```diff
@@ -0,0 +1,142 @@
+  }
+
+  #shareLink {
+    background-color: #fff;
+    border: 1px solid #979797;
+    min-width: 18rem;
+    width: auto;
+  }
+
+  .tooltip-share .tooltip-inner {
+    padding: 5px;
+    width: 150px;
+  }
+
+  .tooltip-share .kudos_msg {
+    padding-left: 2px;
+    padding-right: 2px;
+  }
+
+  .tooltip-inner .title-tooltip {
+    font-size: 0.65rem;
+  }
+
+  @media (max-width: 991.98px) {
+
+    .modal-content {
+      width: 80%;
+    }
+
+    #share-modal p {
+      text-align: center;
+    }
+
+    #share-modal #kudos-image {
+      margin: auto;
+    }
+  }
+
+  @media (max-width: 767.98px) {
+
+    .modal-content {
+      width: 100%;
+    }
+
+    .share-button {
+      margin-bottom: 1rem;
+    }
+  }
+
+  @media (max-width: 575.98px) {
+    #shareLink {
+      min-width: auto;
+      width: 100%;
+      margin-bottom: 1rem;
+    }
+  }
+</style>
+
+<div id="share-modal" class="modal">
+  <div class="modal-content row">
+    <div class="col-12 text-right font-title">
+      <i class="fas fa-times"></i>
+    </div>
+    <div class="col-12 row">
+      <div class="col-12 text-center">
+        <h1 class="font-title-lg mb-4">Share Kudos!</h1>
+      </div>
+      <div id="kudos-image" class="col-12 offset-lg-4 col-lg-4">
+        <img src="{% static kudos.image %}" alt="{{ kudos.name }}" class="w-100">
+      </div>
+      <div class="col-12 col-lg-4 my-auto">
+        <p>
+          <span class="text-color_blue text-weight_semibold">{{ kudos.name|humanize_name }}</span> <br>
+          <span class="text-color_blue text-weight_semibold">{{ kudos.rarity }}</span>
+          ({{ kudos.num_clones_in_wild_counting_indirect_send }} out of {{ kudos.num_gen0_clones_allowed }}) <br>
+          <span id='kudosNumClonesAvailable' class="text-color_blue text-weight_semibold">{{ kudos.num_clones_available_counting_indirect_send }}</span> Total Available <br>
+          <span id='kudosNumClonesAllowed' class="text-color_blue text-weight_semibold">{{ kudos.num_clones_in_wild_counting_indirect_send }}</span> Total minted
+        </p>
+      </div>
+    </div>
+    <div class="col-12 my-4 text-center">
+      <input class="form__input"type="text" value="www.gitcoin.co{{request.get_full_path}}" id="shareLink" readonly>
+      <button class="btn btn-gc-blue button__small font-body" id="copyLink">
+        <i data-toggle="tooltip" title="" data-html="true" 
+          data-original-title="<span class='title-tooltip text-center'>Click To Copy</span>"
+          data-template='<div class="tooltip-kudos tooltip-share" role="tooltip"><div class="tooltip-inner"></div></div>' class="text-color_blue">
+          Copy Link
+        </i>
+      </button>
+    </div>
+    <div class="col-12 text-center">
+      <a class="btn btn-gc-blue share-button font-body" target="_blank" rel="noopener noreferrer"
+        href="https://twitter.com/intent/tweet?text=Check%20out%20the%20{{kudos.name|humanize_name}}%20kudos%21%20www.gitcoin.co{{request.get_full_path}}"
```

Comment:
As we talked about this, we can mount the url as `{{ request.META.HTTP_HOST }}{% url 'kudos_details' kudos.id kudos.name %}` I need to change this in other places too